### DISCUSSION TITLE
Typescript autocomplete with query keys

### DISCUSSION THREAD

**Discussion by nick-0101** [TanStack/query#5394] [Ideas] [answered]
When doing something like: 

`const userOrganizations = useQuery(['userOrgs'], () => getUserOrgs())`

It would be nice to get autocomplete with typescript for the query key when typing 'userOrgs' if the same query is being used in another component. This is a simple example and I know query keys can get really complex with nested objects and stuff but I was just wondering if this has been attempted or decided to be left out as I think it could be valuable when working in a big project with many queries as it gets difficult to remember what query keys are being used.

If it's possible to do I'd be happy to start a branch.

**Comment by TkDodo** [comment, marked as answer]
It's not possible without an up-front definition. Queries can be added lazily, everywhere
